### PR TITLE
Update datastore to query ElasticSearch instead of Mongo

### PR DESCRIPTION
**Pull request by gshaw1997**
This PR implements gateways for ElasticSearch and switches out the Mongo gateways in favor of the Elastic gateways for search and suggestion.

### CODE REVIEW

File: src/Shared/ElasticSearchHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  query: ElasticSearchQuery,
+): Promise<{
+  total: number;
+  outcomes: StandardOutcome[];
+}> {
+  return new Promise<{
+    total: number;
+    outcomes: StandardOutcome[];
+  }>((resolve, reject) => {
+    request({
```
Comment (DonSeannelly) [inline src/Shared/ElasticSearchHelpers.ts]: Why are we making a new promise around something that is already a promise?

File: package.json [hunk 2]
```diff
@@ -32,18 +32,21 @@
...
     "morgan": "^1.9.0",
+    "request-promise": "^4.2.4",
     "spellchecker": "^3.4.4",
     "stemmer": "^1.0.2",
     "stopword": "^0.1.10"
   },
   "devDependencies": {
     "@types/cors": "^2.8.4",
     "@types/dotenv": "^4.0.2",
+    "@types/elasticsearch": "^5.0.32",
```
[the remark this answers is not in the document]
Comment (gshaw1997) [reply package.json]: The ElasticSearch package is not being used. It's types are being used to give the developer information about the response type from making the search request (`SearchResponse`) to ElasticSearch via `request-promise`.

[the remark this answers is not in the document]
Comment (DonSeannelly) [reply package.json]: Ah, gotcha. Neat

File: src/Shared/ElasticSearchHelpers.ts (hunk 1 above)
Comment (gshaw1997) [reply src/Shared/ElasticSearchHelpers.ts]: The `request-promise` library does not return a `Promise`, it returns a `PromiseLike` object being it's internal `Bluebird` type. So just returning the result of the `request` call, the interface contract is broken and Typescript complains.

**Pull request merged**